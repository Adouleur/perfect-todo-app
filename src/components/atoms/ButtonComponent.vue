<script setup>
import { defineProps } from 'vue';

const props = defineProps({
  class: { type: String, default: '' }
});
</script>

<template>
  <button type="button" :class="props.class">
    <slot></slot>
  </button>
</template>

<style scoped>
button {
  padding: 5px 10px;
  border: none;
  border-radius: 4px;
  cursor: pointer;
}
</style>
