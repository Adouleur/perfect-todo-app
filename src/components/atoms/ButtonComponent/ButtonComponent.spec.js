import { mount } from '@vue/test-utils';
import ButtonComponent from '@/components/atoms/ButtonComponent/ButtonComponent.vue';

describe('ButtonComponent', () => {
  let wrapper;
  const testClass = 'test-class';
  const slotContent = 'Click Me!';

  beforeEach(() => {
    wrapper = mount(ButtonComponent, {
      props: {
        class: testClass
      },
      slots: {
        default: slotContent
      }
    });
  });

  it('renders a button with default props', () => {
    expect(wrapper.find('button').exists()).toBe(true);
  });

  it('applies the provided class', () => {
    expect(wrapper.find('button').classes()).toContain(testClass);
  });

  it('renders slot content correctly', () => {
    expect(wrapper.text()).toBe(slotContent);
  });
  it('matches the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
