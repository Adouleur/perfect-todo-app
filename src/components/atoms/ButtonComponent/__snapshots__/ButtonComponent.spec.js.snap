// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ButtonComponent > matches the snapshot 1`] = `"<button data-v-79b848db="" type="button" class="test-class">Click Me!</button>"`;
